dstep/translator/MacroDefinition.d

```diff
@@ -0,0 +1,1404 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Jun 03, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+module dstep.translator.MacroDefinition;
+
+import std.array : Appender;
+
+import clang.c.Index;
+import clang.Cursor;
+import clang.Token;
+
+import dstep.translator.Context;
+import dstep.translator.Output;
+
+bool accept(Args...)(ref TokenRange tokens, ref string spelling, TokenKind kind)
+{
+    if (!tokens.empty && tokens.front.kind == kind)
+    {
+        foreach (arg; Args)
+        {
+            if (tokens.front.spelling == arg)
+            {
+                tokens = tokens[1..$];
+                spelling = arg;
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+bool accept(Args...)(ref TokenRange tokens, TokenKind kind)
+{
+    if (!tokens.empty && tokens.front.kind == kind)
+    {
+        foreach (arg; Args)
+        {
+            if (tokens.front.spelling == arg)
+            {
+                tokens = tokens[1..$];
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+bool accept(ref TokenRange tokens, ref string spelling, TokenKind kind)
```

Comment:
If you don't specify any template parameters (since it's a variadic template parameter) it should cause a conflict. If the intention is that at least one template parameter should always be passed, then add a template constraint to enforce that.